### PR TITLE
Add --less-search-pattern option to tell less to search for pattern

### PR DESCRIPTION
**Pull request by HarrisonMc555**
The `less` pager can search for a pattern. In addition, you can instruct it to
search for a given pattern on startup by passing in the `-p` flag followed by
the pattern to search for. Since the `batgrep` command is searching for a
pattern and invoking the pager, it has the opportunity to take advantage of this
functionality.

This patch adds a `-p` flag to `batgrep` which, in turn, passes `-p $PATTERN` to
the pager invoked. This may be specific to `less`, which is
unfortunate. However, it doesn't seem like there would be a convenient way to
make this more general. In addition, most pagers forward the actual paging to
`less` anyways.

To clarify that the behavior is specific to `less`, the long name is
`--less-search-pattern`.

With all that being said, I would appreciate any feedback or discussion! This is
simply my proposed first draft, and I'm more than willing to rename, rearrange,
etc. And, if you don't want this in the project, that's totally
understandable. I just may keep my own personal patch on my own computer :)

### CODE REVIEW

**Comment by HarrisonMc555** [comment]
I just found a bug when using `--less-search-pattern` with ripgrep's `--fixed-strings` option where `less` would still treat the pattern as a regular expression.

This is now fixed in a new commit.

**Comment by eth-p** [comment]
Thanks for the pull request! I left a couple of recommendations/suggestions, and be happy to merge this after resolving those.

If you want a way to detect if the pager would be `less` so you can add a safety check, I could add a function (`pager_is_less`?) to the pager library for that, too.

File: src/batgrep.sh [hunk 1]
```diff
@@ -134,6 +139,20 @@ if [[ "$OPT_CONTEXT_BEFORE" -eq 0 && "$OPT_CONTEXT_AFTER" -eq 0 ]]; then
...
 
+if [[ "$OPT_LESS_SEARCH_PATTERN" = true ]]; then
+	if [[ "$OPT_FIXED_STRINGS" = true ]]; then
+		# Depending on your editor, this may look identical. However, there is a
+		# special control character at the beginning of the pattern string, right
+		# after the first double quote. It is a a ^R, or Control-R, character. This
+		# instructs less to NOT use regular expressions, which is what the -F flag
+		# does for ripgrep. If we did not use this, then less would match a
+		# different pattern than ripgrep searched for.
+		SCRIPT_PAGER_ARGS+=(-p "$PATTERN")
```
[the remark this answers is not in the document]
Comment (HarrisonMc555) [reply src/batgrep.sh]: Great idea! It's such an uncommon occurrence I forget that bash has a literal syntax for them.

File: src/batgrep.sh [hunk 2]
```diff
@@ -134,6 +139,20 @@ if [[ "$OPT_CONTEXT_BEFORE" -eq 0 && "$OPT_CONTEXT_AFTER" -eq 0 ]]; then
 	OPT_HIGHLIGHT=false
 fi
 
+if [[ "$OPT_LESS_SEARCH_PATTERN" = true ]]; then
```
[the remark this answers is not in the document]
Comment (HarrisonMc555) [reply src/batgrep.sh]: Oh I hadn't noticed, thanks for pointing that out.

**Comment by eth-p** [comment]
> Would it be a good idea to check for less? The only real way to do that would be to check the name of the command.

That's pretty much how both `bat` and the pager library/script check for it.

> If a user had another script that passed things along to less I feel like it could be annoying if they couldn't use it because we were checking for a hardcoded less string.

Good point. This could be an issue... A possible solution would be to execute `less --version` and grep the output? I feel like that would be fairly reliable, even with a wrapper script.

> On the flip side, it could be frustrating to figure out why the flag wasn't working if you were using a different pager. 

That was my thought, yeah.

> That's why I tried to be really specific in the flag name and description.

If we're able to detect `less` support, you should be able to change the name to be more general, and just print an error message if the pager isn't supported.

> I'm leaning more towards documenting the fact that it's specific to less but not checking. If you feel strongly the other way, though, I'm willing to concede the point.

I would prefer the other way around, to be honest. It would allow for the possibility of adding support for other pagers (if anybody needs it) without having to introduce an additional option or make a breaking change.

**Comment by eth-p** [comment]
I had to fix some merge conflicts that I introduced (by adding `pager_name` in myself earlier), but it's good to go now!

**Comment by eth-p** [comment]
Thank you for your contribution! Additionally, another "thanks" because the PR also helped me realize that I was forgetting to pass `--terminal-width` to `bat` when using a pager.

**Comment by HarrisonMc555** [comment]
No problem! Glad it helped.

**Pull request merged**